src/ros_filter.cpp

```diff
@@ -813,6 +813,18 @@ namespace RobotLocalization
     nhLocal_.param("dynamic_process_noise_covariance", dynamicProcessNoiseCovariance, false);
     filter_.setUseDynamicProcessNoiseCovariance(dynamicProcessNoiseCovariance);
 
+    std::vector<double> initial_state(STATE_SIZE, 0.0);
+    if(nhLocal_.getParam("initial_state", initial_state))
+    {
+      if(initial_state.size() != STATE_SIZE)
+      {
+        ROS_ERROR_STREAM("Initial state must be of size " << STATE_SIZE << ". Provided config was of "
+          "size " << initial_state.size() << ". The initial state will be ignored.");
```

Comment:
Does `filter_.setState` silently reject incorrectly-sized states? Or should lines 825 & 826 be in an else block?